### PR TITLE
chore: pre-commit hook for testing & linting

### LINKED ISSUE #35: feat: Pre-commit hooks for linting and tests
**Is your feature request related to a problem? Please describe.**

Currently, there is a lack of testing and linting enforcement before committing code, which can result in unsupervised code being pushed and have detrimental effects on code quality and the code review process.

**Describe the solution you'd like**

I propose adding a pre-commit hook that includes running test scripts and performing linting checks on the codebase. To optimize the linting process for the growing codebase, we can leverage the 'lint-staged' npm package. 'lint-staged' enables us to selectively lint and fix only the files that are staged for commit, reducing potential delays caused by the entire codebase's linting. By implementing this pre-commit hook, we ensure that testing and linting are enforced before code is committed, leading to improved code quality and a more effective code review process

**Acknowledgement**

> We thank you for your valuable contribution and would like to acknowledge it
> on our [contributors][contributors] page. To help us with that, we would like
> to kindly request that you help us with this by commenting on this issue with
> a request to the [@all-contributors bot][all-contributors-bot] of the form
> `@all-contributors please add @YOUR_GH_HANDLE for TYPE_1, TYPE_2, ...`, where
> `TYPE_1` etc. refer to [contribution types supported by the All Contributors
> Specification][all-contributors-types], e.g., `idea`.

[all-contributors-bot]: <https://allcontributors.org/docs/en/bot/overview>
[all-contributors-types]: <https://allcontributors.org/docs/en/emoji-key>
[contributors]: ../../CONTRIBUTORS

### PR DESCRIPTION
**Pull request by JaeAeich**
This commit ensures that all the tests have passed and the files have been linted before comitting, ensuring a much smoother development process and less error prone commits.

fix #35

## Description
Having all the files auto linted and tested before commit reduces chances of errors in the commit.

-Fixes #35 

## Checklist

- [x] My code follows the [contributing guidelines][contributing] of this
     project, including, in particular, with regard to any style guidelines
- [x] The title of my PR complies with the [Conventional Commits
      specification][conventional-commits]; in particular, it clearly indicates
      that a change is a breaking change
- [x] I am aware that all my commits will be squashed into a single commit,
      using the PR title as the commit message.
- [x] I have performed a self-review of my own code
- [x] I have commented my code in hard-to-understand areas
- [x] I have updated the user-facing documentation to describe any new or
      changed behavior
- [ ] I have added type annotations for all function/class/method interfaces
      or updated existing ones
- [ ] I have provided [JSDoc][jsdoc] block tags for all
      functions/classes/methods or updated existing ones
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature
      works
- [x] New and existing unit tests pass locally with my changes
- [x] I have not reduced the existing code coverage
- [ ] I have asked the [@all-contributors bot][all-contributors-bot] to
      acknowledge my contributions by commenting on this PR with a request of
      the form `@all-contributors please add @YOUR_GH_HANDLE for TYPE_1,
    TYPE_2, ...`, where `TYPE_1` etc. refer to [contribution types supported
      by the All Contributors Specification][all-contributors-types] OR I do
      not want my contributions to be acknowledged

[all-contributors-bot]: https://allcontributors.org/docs/en/bot/overview
[all-contributors-types]: https://allcontributors.org/docs/en/emoji-key
[contributing]: CONTRIBUTING.md
[conventional-commits]: https://www.conventionalcommits.org/en/v1.0.0/
[jsdoc]: https://jsdoc.app/index.html

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -57,6 +57,25 @@ This will start the development server on your local machine and allow you to st
...
+   ```
+   npm run lint
+   ```
+
+5. Run all test using the following command.
+
+   ```
+   npm run test
+   ``` 
+   Note : Since packages have interdependency, you need to run `dev` script atleast once before running `lint` and `test` scripts globally.
```
Comment (uniqueg) [inline README.md:71]: Please add this as a markdown _comment_ with the `>` syntax. Use comments elsewhere as well, if applicable.

Also, I think you can then either omit the word "Note" altogether, or at least remove the whitespace _before_ the colon.

File: README.md [hunk 2]
```diff
@@ -57,6 +57,25 @@ This will start the development server on your local machine and allow you to st
...
+
+Above commands are global commands, they run for the entire monorepo which includes all the pacakges, apps and docs etc but these commands can be coupled with `-w` or `-workspace` flag to run them on a specific repo/workspace inside this monorepo.
+
+for example, to run test script on the package 'elixir-ui', use the following command:
```
Comment (uniqueg) [inline README.md:75]: "For example", with capital "F"

**Review by uniqueg — CHANGES_REQUESTED** [review]
Very minor changes, otherwise LGTM 👍

File: README.md [hunk 3]
```diff
@@ -57,6 +57,27 @@ This will start the development server on your local machine and allow you to st
...
+
+   ```
+   npm run test
+   ``` 
+> Note: Since packages have interdependency, you need to run `dev` script atleast once before running `lint` and `test` scripts globally.
+
+<a id="point-6"></a>
+
+>Above commands are global commands, they run for the entire monorepo which includes all the packages, apps and docs etc but these commands can be coupled with `-w` or `-workspace` flag to run them on a specific repo/workspace inside this monorepo.
+1. To run test script on the package `elixir-ui`, use the following command:
```
Comment (git-anurag-hub) [inline README.md:76]: Point 6.

**Review by git-anurag-hub — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (git-anurag-hub) [reply README.md:76]: Change this and please merge @JaeAeich!

Comment (JaeAeich) [reply README.md:76]: can you please elaborate on what to change @git-anurag-hub, do I have to remove point 6?

**Review by uniqueg — APPROVED** [review]
Looks good from my side

**Comment by uniqueg** [comment]
But note that the PR title did not conform to the <= 50 chars rule ;-) I've fixed that

**Pull request merged**